### DISCUSSION TITLE
Lighthouse system trajectory generate

### DISCUSSION THREAD

**Discussion by nbpilot007** [bitcraze/discussions#1477] [Q&A]
Hey, I have recently set up the light house positioning deck and have localized cf using the cf client. Now that this is done I want to run trajectories on it, what is the right way to start with it also is there a way to run parametric trajectories?

Thank you

**Comment by knmcguire** [comment]
Great! So the cfclient is a bit limited in the sense of trajectory upload, but I recommend you to get more familiar with its backend instead, namely the crazyflie python library. The lighthouse positioning is completely onboard so you don't need to be connected to the cfclient at the same time (and that wouldn't be possible anyway from 1 crazyradio).

There are some step-by-step guides for some basic guides of how to work with the lib: https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/user-guides/sbs_connect_log_param/

Once you have done that, you can take a look at the high level commander api: https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/crazyflie/high_level_commander/

You can also find some examples on the python library how to uploade these trajectories like the figure8 https://github.com/bitcraze/crazyflie-lib-python/blob/master/examples/autonomy/autonomous_sequence_high_level.py

**Comment by nbpilot007** [comment]
Hey, is there a way we can give (x,y,z) with roll pitch and yaw to the crazyfly?

**Reply by Bananazooka** [reply]
Hello, 
Please check out the full state setpoint in the commander [here](https://www.bitcraze.io/documentation/repository/crazyflie-lib-python/master/api/cflib/crazyflie/commander/)

For any followup questions, could you please write a new thread with this issue?
